src/ui/buttons/buttonDesigns/divideLogoTopTextBottomAnimation.jsx

```diff
@@ -0,0 +1,163 @@
+    cssClasses: {
+        DOM_READY:                  CLASS.DOM_READY,
+        ANIMATION_CONTAINER:        'personalized-design-container',
+        PAYPAL_LOGO:                LOGO_CLASS.LOGO,
+        ANIMATION_LABEL_CONTAINER:  'personalized-label-container',
+        PAYPAL_BUTTON:              CLASS.BUTTON
+    }
+};
+
+export function LogoTopTextBottomComponent({ designLabelText } : ContentOptions) : ChildType {
+    const CONTAINER_CLASS = LOGO_TOP_TEXT_BOTTOM_CONFIG.cssClasses.ANIMATION_CONTAINER;
+    const LABEL_CLASS = LOGO_TOP_TEXT_BOTTOM_CONFIG.cssClasses.ANIMATION_LABEL_CONTAINER;
+
+    return (
+        <Fragment>
+            <div class={ LABEL_CLASS } data-design-experiment='104530'> <span>{designLabelText}</span></div>
+            <style innerHTML={ `
+                .${ CLASS.DOM_READY } .${ CONTAINER_CLASS } img.${ LOGO_CLASS.LOGO }{
+                    position: relative;
+                }
+                
+                .${ CONTAINER_CLASS } .${ LABEL_CLASS } {
+                    opacity: 0;
+                }
+            ` } />;
+        </Fragment>
+    );
+}
+
+// Returns label container if the button sizes match
+export const getLogoTopTextBottomProps = function(document : Object, configuration : Object) : AnimationProps | null {
+    const { ANIMATION_CONTAINER, ANIMATION_LABEL_CONTAINER, PAYPAL_BUTTON, PAYPAL_LOGO } = configuration.cssClasses;
+    const { min, smallMax, max } = configuration;
+    let animationConfig = null;
+
+    // get the animation main container to force specificity( in css ) and make sure we are running the right animation
+    const designContainer = (document && document.querySelector(`.${ ANIMATION_CONTAINER }`)) || null;
+    if (!designContainer) {
+        return null;
+    }
+
+    // return null if animation should not be played for the button size
+    const animationContainerWidth = designContainer.offsetWidth;
+    if (animationContainerWidth < min || animationContainerWidth > max) {
+        // remove label element from dom
+        designContainer.querySelector(`.${ ANIMATION_LABEL_CONTAINER }`).remove();
+        return null;
+    }
+
+    // get the label container that animation will be applied to
+    const paypalButtonContainerElement = designContainer.querySelector(`.${ PAYPAL_BUTTON }`) || null;
+    // get starting position for element so it doesn't jump when animation begins
+    const paypalLogoElement = (paypalButtonContainerElement && paypalButtonContainerElement.querySelector(`.${ PAYPAL_LOGO }`)) || null;
+
+    // get correct configuration according to the button size (small or medium)
+    if (paypalLogoElement && paypalButtonContainerElement) {
+        animationConfig = animationContainerWidth < smallMax ?
+            { paypalLogoTopPosition: 25, labelTextFontSize: 7 } : { paypalLogoTopPosition: 30, labelTextFontSize: 9 };
+    } else {
+        return null;
+    }
+   
+    return {
+        designContainer,
+        paypalButtonContainerElement,
```

Comment:
there is no need to create another object, you can just destructure this object in here, in order to just return the properties you need

Reply:
the object has two properties, `paypalLogoTopPosition and labelTextFontSize`, if I do that, we are changing one line to two lines and adding more variables in the next function which will destructure this object. I don't know if this is the best way to clean the code

Reply:
well we just have two properties so I dont see the need for that, but I think that is personal way to do it. I am okay with what you decide.